### ISSUE TITLE
FloatScaled concepts

### ISSUE THREAD

**Issue by Noggog** [Mutagen-Modding/Mutagen#181]
Right now some floats need scaling from their on disk value to one that humans can deal with more easily.

Rotations are an example of this.

Problem is that converting to/from values means sometimes the values change slightly.  This throws off modification detection systems, such as xedit override detection, etc.

Can perhaps upgrade floats /w scaling to be a more complex struct, such as:
```
public struct ScaledFloat
{
    public readonly float? PreScaledValue;
    public readonly float ScaledValue;

    public static implicit operator ScaledFloat(float f)
    {
         return new ScaledFloat()
         {
             ScaledValue = f;
         }
    }
}
```

During export, it would use the value in `PreScaledValue` as the value to write, which would mean the scaled value isn't even used, preventing any jitter.   Only if a user modified the value would the `PreScaledValue` be wiped, and the scaling would be applied to the user given value.

This would keep unmodified floats much more stable

**Comment by mrudat** [comment]
Wouldn't the general pattern (for dotnet) be to do something like:
```C#
public static readonly theScaleValueForThisFloat = 0.1234F;

private float thisFloat;

public float ThisFloat
{
  get
  {
    return thisFloat * theScaleValueForThisFloat;
  }
  set
  {
    thisFloat = value / theScaleValueForThisFloat;
  }
}
```

**Comment by Noggog** [comment]
Yeah, that's pretty normal/viable, too.   Only downside to the approach is if the user is going to do repeated interactions with a single property, it'll incur multiple scaling passes.   The speed loss isn't the problem, there, but rather potential accumulating rounding errors, I think?

The ScaledFloat would prevent this, as the ScaledValue can be set/used multiple times without any scaling being applied.  Only on final write out does the scaling get applied once, with hopefully minimal rounding errors.
